Packs/FeedMandiant/Integrations/FeedMandiant/FeedMandiant.py

```diff
@@ -0,0 +1,692 @@
+        self._token = resp.get('access_token')
+
+        integration_context = get_integration_context()
+        integration_context.update({'token': self._token})
+        # Add 10 minutes buffer for the token
+        integration_context.update({'valid_until': datetime.timestamp(datetime.now(timezone.utc)) - 600})
+        set_integration_context(integration_context)
+
+        return self._token
+
+    def get_metadata(self, indicator: Dict, indicator_type: str) -> Dict:
+        """
+        Get additional information for given indicator.
+
+        Args:
+            indicator (Dict): Basic indicator.
+            indicator_type (str): The indicator type.
+        Returns:
+            Dict: Additional metadata of the indicator.
+        """
+        try:
+            identifier = indicator.get('id')
+            res = self._http_request(
+                method="GET", url_suffix=f'/v4/{MAP_TYPE_TO_URL[indicator_type]}/{identifier}', timeout=self._timeout,
+                ok_codes=[200])
+        except DemistoException:
+            res = indicator
+        return res
+
+    def get_indicator_info(self, identifier: str, info_type: str) -> List:
+        """
+        Get additional information for given indicator.
+
+        Args:
+            identifier (str): Indicator's id.
+            info_type (str): The indicator type.
+        Returns:
+            Dict: Additional metadata of the indicator.
+        """
+        try:
+            res = self._http_request(method="GET", url_suffix=f'/v4/malware/{identifier}/{info_type}',
+                                     timeout=self._timeout, ok_codes=[200])
+            if info_type == 'attack-pattern':
+                res = res.get('attack-patterns')
+
+                if res.get('malware', {}).get('attack-patterns') == 'redacted':
+                    res = []
+            else:
+                res = res.get(info_type, [])
+
+        except DemistoException:
+            res = []
+        return res
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_new_indicators(client: MandiantClient, last_run: str, indicator_type: str, limit: int) -> List:
+    """
+    Get new indicators list.
+    Args:
+        client (MandiantClient): client
+        last_run (str): last run as free text or date format
+        indicator_type (str): the desired type to fetch
+        limit (int): number of indicator to fetch
+    Returns:
+        List: new indicators
+
+    """
+    start_date = dateparser.parse(last_run, settings={'TIMEZONE': 'UTC'})
+
+    if indicator_type == 'Indicators':
+        # for indicator type the earliest time to fetch is 90 days ago
+        earliest_fetch = dateparser.parse('90 days ago', settings={'TIMEZONE': 'UTC'})
+        start_date = max(earliest_fetch, start_date)
+
+    params = {'start_epoch': int(start_date.timestamp()), 'limit': limit} if indicator_type == 'Indicators' else {}
```

Comment:
Yes, I've checked with our contact in Mandiant, currently filtration using date is only available for indicators and he doesn't know if it will be available for the other types in the future.